### ISSUE TITLE
[v4 to v5] How to attach cucumber scenario logs to allure reports ?

### ISSUE THREAD

**Issue by Kignuf** [webdriverio/webdriverio#4183]
**Describe your question with as much detail as possible**
Hello, I am migrating from v4 to v5, using cucumber framework.
In v4, it was possible to provide a node writable stream to webdriverio and that is where logs would be written ([see doc here](http://v4.webdriver.io/guide/getstarted/configuration.html#logOutput)). I have implemented a writable stream that saves logs into separate files for each scenario within a given feature.
At the end of each scenario, I attach those client logs to the allure report.

Since v5 has revamped the logging system quite a lot and does not seem to allow the use of a writable stream anymore, I was wondering if someone can give me some advices to achieve the same result.

**If it's about a specific piece of code, try and include some of it to support your question.**
V5 does not accept writable streams for outputDir (~ logOutput in v4):
```
// packages/wdio-cli/src/launcher.js
if (config.outputDir) {
    fs.ensureDirSync(path.join(config.outputDir))
    process.env.WDIO_LOG_PATH = path.join(config.outputDir, 'wdio.log')
}
```

**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.11.0
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 10.15.3
 - **NPM version:** 6.10.0
 - **Browser name and version:** appium 1.13.0
 - **Platform name and version:** MacOSX 10.14.5
 - **Additional wdio packages used (if applicable):** @wdio/allure-reporter @wdio/cucumber-framework

**Additional context**
If changes to webdriverio are involved, I can do the coding :)

**Comment by christian-bromann** [comment]
> I was wondering if someone can give me some advices to achieve the same result.

There is currently no way to attach logs to an existing stream.

**Comment by Kignuf** [comment]
Thanks for the info. What I meant is, maybe there is a way to save my logs into separate files for each scenario, but implemented completelly differently.

I am struggling a bit to understand how webdriverio works in regards to the logging mechanism.
When I set my outputDir to undefined I do get a rather neat logging to the console, similar to v4. However when it set that to a path, my client logs (wdio-0-0.log) are very messy and do not look like my terminal output at all.

This could be a bug, if so I'll make a bug report with a repo to reproduce.

**Comment by christian-bromann** [comment]
> are very messy

What is messy? We create a log file for every worker.

> This could be a bug, if so I'll make a bug report with a repo to reproduce.

Nope. This is the designed behavior.

**Comment by Kignuf** [comment]
Sorry I think my message was a bit missleading.
I'll make a proper bug report to explain why I find the logs inside wdio-X-X.log file to be hard to read and "messy".

I've been through webdriverio code and I'm starting to grasp a bit better how it's working, especially the role of @wdio/logger and the fact that it is currently not possible to alter its behavior. Indeed I feel a but stuck with my problem at the moment, but I want to persevere because having my scenario logs neatly attached into each scenario within allure report is really a must have for me.

Thanks again for taking the time to answer :)

**Comment by christian-bromann** [comment]
> but I want to persevere because having my scenario logs neatly attached into each scenario within allure report is really a must have for me.

We are happy to accept PRs if they help the community with their testing needs.

Are you gonna update this or can this be closed?

**Comment by Kignuf** [comment]
Hello Christian, I'm closing the ticket. Thanks again for your time.
Btw, I updated to latest releases of all the wdio packages and it seems my log issues are fixed.
Also I finally understood that I got confused about the fact that calls to session/<sessionid>/log are logged in client logs, but the response, which contains the server/logcat logs, is also logged. So in fact, my client logs also contain server and logcat logs.
